lib/fog/azurerm/models/network/virtual_network.rb

```diff
@@ -8,37 +8,70 @@ class VirtualNetwork < Fog::Model
         attribute :id
         attribute :location
         attribute :resource_group
-        attribute :dns_list
-        attribute :subnet_address_list
-        attribute :network_address_list
+        attribute :dns_servers
+        attribute :subnets
+        attribute :address_prefixes
 
         def self.parse(vnet)
           hash = {}
           hash['id'] = vnet['id']
           hash['name'] = vnet['name']
           hash['resource_group'] = vnet['id'].split('/')[4]
           hash['location'] = vnet['location']
-          hash['dns_list'] = vnet['properties']['dhcpOptions']['dnsServers'].join(',') unless vnet['properties']['dhcpOptions'].nil?
-          hash['network_address_list'] = vnet['properties']['addressSpace']['addressPrefixes'].join(',') unless vnet['properties']['addressSpace']['addressPrefixes'].nil?
-          subnet_address_list = []
+          hash['dns_servers'] = vnet['properties']['dhcpOptions']['dnsServers'] unless vnet['properties']['dhcpOptions'].nil?
+          hash['address_prefixes'] = vnet['properties']['addressSpace']['addressPrefixes'] unless vnet['properties']['addressSpace']['addressPrefixes'].nil?
+
+          subnets = []
           vnet['properties']['subnets'].each do |subnet|
-            subnet_address_list << subnet['properties']['addressPrefix']
+            subnet_object = Fog::Network::AzureRM::Subnet.new
+            subnets << subnet_object.merge_attributes(Fog::Network::AzureRM::Subnet.parse(subnet))
```

Comment:
Plz confirm if we can use subnets.push. This will increase code readability unless rubocop complaints about this.